StRoot/StSpinPool/StFcsPi0FinderForEcal/StFcsPi0FinderForEcal.cxx

```diff
@@ -0,0 +1,507 @@
+      h1list_NEtower[i] = new TH1F(name_hist, title_hist, 200, 0, 20);
+      h1list_NEtower[i]->SetXTitle("Energy [GeV]");
+      sprintf(name_hist, "SEtower_%i", i);
+      sprintf(title_hist, "South %i tower energy spectrum", i + 1);
+      h1list_SEtower[i] = new TH1F(name_hist, title_hist, 200, 0, 20);
+      h1list_SEtower[i]->SetXTitle("Energy [GeV]");
+   }
+
+   h2_EcalMult_vs_TofMult = new TH2F("h2_EcalMult_vs_TofMult", "Ecal Mult vs Tof Mult", 200, 0, 200, 200, 0, 200);
+   h2_EcalMult_vs_TofMult->SetXTitle("Tof Multiplicity");
+   h2_EcalMult_vs_TofMult->SetYTitle("Ecal Multiplicity");
+   h2_cluster_position = new TH2F("h2_cluster_position", "cluster_position", 300, -150, 150, 300, -150, 150);
+   h2_cluster_position->SetXTitle("x [cm]");
+   h2_cluster_position->SetYTitle("y [cm]");
+   h2_cluster_position_cut = new TH2F("h2_cluster_position_cut", "cluster position after cut", 400, -200, 200, 350, -150, 150);
+   h2_cluster_position_cut->SetXTitle("x [cm]");
+   h2_cluster_position_cut->SetYTitle("y [cm]");
+   h2_point_position = new TH2F("h2_point_position", "point_position", 300, -150, 150, 300, -150, 150);
+   h2_point_position->SetXTitle("x [cm]");
+   h2_point_position->SetYTitle("y [cm]");
+
+   h2_cluster_invmass_vs_dgg = new TH2F("h2_cluster_invmass_vs_dgg", "invariant mass vs d_{gg} (cluster)", 60, 0, 60, 60, 0, 0.3);
+   h2_cluster_invmass_vs_dgg->SetXTitle("d_{gg} [cm]");
+   h2_cluster_invmass_vs_dgg->SetYTitle("m_{gg} [GeV]");
+   h2_cluster_invmass_vs_Zgg = new TH2F("h2_cluster_invmass_vs_Zgg", "invariant mass vs Z_{gg} (cluster)", 35, 0, 0.7, 60, 0, 0.3);
+   h2_cluster_invmass_vs_Zgg->SetXTitle("Z_{gg} ");
+   h2_cluster_invmass_vs_Zgg->SetYTitle("m_{gg} [GeV]");
+   h2_point_invmass_vs_dgg = new TH2F("h2_point_invmass_vs_dgg", "invariant mass vs d_{gg} (cluster)", 60, 0, 60, 60, 0, 0.6);
+   h2_point_invmass_vs_dgg->SetXTitle("d_{gg} [cm]");
+   h2_point_invmass_vs_dgg->SetYTitle("m_{gg} [GeV]");
+   h2_point_invmass_vs_Zgg = new TH2F("h2_point_invmass_vs_Zgg", "invariant mass vs Z_{gg} (cluster)", 35, 0, 0.7, 60, 0, 0.6);
+   h2_point_invmass_vs_Zgg->SetXTitle("Z_{gg} ");
+   h2_point_invmass_vs_Zgg->SetYTitle("m_{gg} [GeV]");
+   h2_cluster_dgg_vs_E1pE2 = new TH2F("h2_cluster_dgg_vs_E1pE2", "2 cluster distance vs total energy for FCS ECal cluster pair", 20, 0, 20, 60, 0, 60);
+   h2_cluster_dgg_vs_E1pE2->SetXTitle("E_{1}+E_{2} [GeV]");
+   h2_cluster_dgg_vs_E1pE2->SetYTitle("d_{gg} [cm]");
+   h2_point_dgg_vs_E1pE2 = new TH2F("h2_point_dgg_vs_E1pE2", "energy asymmetry vs total energy for FCS ECal point pair", 20, 0, 20, 60, 0, 60);
+   h2_point_dgg_vs_E1pE2->SetXTitle("E_{1}+E_{2} [GeV]");
+   h2_point_dgg_vs_E1pE2->SetYTitle("d_{gg} [cm]");
+
+   return kStOK;
+}
+
+//-----------------------
+Int_t StFcsPi0FinderForEcal::Finish() {
+   if (filename.length() == 0) return kStOk;
+   const char* fn = filename.c_str();
+   TFile* MyFile = new TFile(fn, "RECREATE");
```

Comment:
This leaks memory. Consider switching to
```suggestion
   TFile MyFile(fn, "RECREATE");
```
or using `std::unique_ptr<TFile>`

Reply:
Yes. But it is in Finish() not in Make(). Still good idea.